src/emit/mod.rs

```diff
@@ -4077,7 +4077,8 @@ pub trait TargetRuntime<'a> {
                         let address = if let Some(address) = address {
                             let address = self.expression(bin, address, &w.vars, function, ns);
 
-                            let addr = bin.builder.build_array_alloca(
+                            let addr = bin.build_array_alloca(
+                                function,
```

Comment:
What is the implication of this?

Reply:
On Solana, llvm requires that all stack allocations happen in the first block of a function. This is a limitation of the BPF backend in llvm.

The stack allocator in the llvm backend is very limited. Stack variables with non-overlapping life times do not reuse stack space, so if you have too many local variables then you overflow the 4KiB stack size (per-frame).

Reply:
Without this change, compiling spl_token.sol crashes llvm with an assert